DependencyInjection/DoctrineExtension.php

```diff
@@ -272,7 +273,8 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
             ->setArguments([
                 $options,
                 new Reference(sprintf('doctrine.dbal.%s_connection.configuration', $name)),
-                new Reference(sprintf('doctrine.dbal.%s_connection.event_manager', $name)),
+                // event manager is only supported on DBAL < 4
+                class_exists(DBALException::class) ? new Reference(sprintf('doctrine.dbal.%s_connection.event_manager', $name)) : null,
```

Comment:
A more accurate feature detection might be checking if `Connection::getEventManager()` exists.